cpp/cmake_modules/Usevcpkg.cmake

```
@@ -0,0 +1,214 @@
+    get_filename_component(_VCPKG_BUILDSYSTEMS_DIR "${CMAKE_TOOLCHAIN_FILE}" DIRECTORY)
+    get_filename_component(VCPKG_ROOT "${_VCPKG_BUILDSYSTEMS_DIR}/../.." ABSOLUTE)
+  else()
+    message(
+      FATAL_ERROR
+        "vcpkg toolchain file not found at path specified in -DCMAKE_TOOLCHAIN_FILE")
+  endif()
+else()
+  if(DEFINED VCPKG_ROOT)
+    # Get it from the CMake variable VCPKG_ROOT
+    find_program(_VCPKG_BIN vcpkg PATHS "${VCPKG_ROOT}" NO_DEFAULT_PATH)
+    if(NOT _VCPKG_BIN)
+      message(FATAL_ERROR "vcpkg not found in directory specified in -DVCPKG_ROOT")
+    endif()
+  elseif(DEFINED ENV{VCPKG_ROOT})
+    # Get it from the environment variable VCPKG_ROOT
+    set(VCPKG_ROOT ENV{VCPKG_ROOT})
+    find_program(_VCPKG_BIN vcpkg PATHS "${VCPKG_ROOT}" NO_DEFAULT_PATH)
+    if(NOT _VCPKG_BIN)
+      message(
+        FATAL_ERROR "vcpkg not found in directory in environment variable VCPKG_ROOT")
+    endif()
+  else()
+    # Get it from the file vcpkg.path.txt
+    find_program(_VCPKG_BIN vcpkg)
+    if(_VCPKG_BIN)
+      get_filename_component(_VCPKG_REAL_BIN "${_VCPKG_BIN}" REALPATH)
+      get_filename_component(VCPKG_ROOT "${_VCPKG_REAL_BIN}" DIRECTORY)
+    else()
+      if(CMAKE_HOST_WIN32)
+        set(_VCPKG_PATH_TXT "$ENV{LOCALAPPDATA}/vcpkg/vcpkg.path.txt")
+      else()
+        set(_VCPKG_PATH_TXT "$ENV{HOME}/.vcpkg/vcpkg.path.txt")
+      endif()
+      if(EXISTS "${_VCPKG_PATH_TXT}")
+        file(STRINGS "${_VCPKG_PATH_TXT}" VCPKG_ROOT)
+      else()
+        message(
+          FATAL_ERROR
+            "vcpkg not found. Install vcpkg if not installed, "
+            "then run vcpkg integrate install or set environment variable VCPKG_ROOT.")
+      endif()
+      find_program(_VCPKG_BIN vcpkg PATHS "${VCPKG_ROOT}" NO_DEFAULT_PATH)
+      if(NOT _VCPKG_BIN)
+        message(FATAL_ERROR "vcpkg not found. Re-run vcpkg integrate install "
+                            "or set environment variable VCPKG_ROOT.")
+      endif()
+    endif()
+  endif()
+  set(CMAKE_TOOLCHAIN_FILE
+      "${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+      CACHE STRING "Path to vcpkg CMake toolchain file")
+endif()
+message(STATUS "Using CMAKE_TOOLCHAIN_FILE: ${CMAKE_TOOLCHAIN_FILE}")
+message(STATUS "Using VCPKG_ROOT: ${VCPKG_ROOT}")
+
+# ----------------------------------------------------------------------
+# Get VCPKG_TARGET_TRIPLET
+
+if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
+  set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}")
+endif()
+# Explicitly set manifest mode on if it is not set and vcpkg.json exists
+if(NOT DEFINED VCPKG_MANIFEST_MODE AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/vcpkg.json")
+  set(VCPKG_MANIFEST_MODE ON CACHE BOOL "Use vcpkg.json manifest")
+  message(STATUS "vcpkg.json manifest found. Using VCPKG_MANIFEST_MODE: ON")
+endif()
+# vcpkg can install packages in two different places
+set(_INST_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/vcpkg_installed") # try here first
```

```suggestion
set(_INST_SOURCE_DIR "${CMAKE_CURRENT_BUILD_DIR}/vcpkg_installed") # try here first
```

I don't think that changing source directory is a good idea.

---

Thank you, this comment was very helpful because it made me realize that this script actually needs to look in _both_ of these places `${CMAKE_CURRENT_SOURCE_DIR}/vcpkg_installed` and `${CMAKE_CURRENT_BUILD_DIR}/vcpkg_installed` because of an inconsistency in vcpkg, as described here: https://github.com/microsoft/vcpkg/issues/14418#issuecomment-726467751

---

Implemented this change in aeda9041d161ca3fdb3c46e6d1c8b5b99847c3a2

---

Whoops, this is supposed to be `CMAKE_CURRENT_BINARY_DIR` not `CMAKE_CURRENT_BUILD_DIR`

---

Fixed in 8543407dd614a38cd0a577063e8d790cc07e044f

---

Oh, sorry.